src/ebonite/ext/prebuild.py

```diff
@@ -0,0 +1,62 @@
+import os
+import re
+import shutil
+import sys
+import tempfile
+
+from jinja2 import Environment, FileSystemLoader
+
+from ebonite.build.docker import (RemoteDockerRegistry, create_docker_client, image_exists_at_dockerhub,
+                                  login_to_registry, repository_tags_at_dockerhub)
+from ebonite.utils.log import logger
+
+
+def prebuild_image(prebuild_path, name_template, python_version, *, push=False):
+    tag = name_template.format(python_version)
+    if image_exists_at_dockerhub(tag):
```

Comment:
maybe pull it here?

Reply:
Do you mean pulling `tag` image locally?

I think it is a bad idea. Running this script (as main) would currently pull you ~20 large images (Python+Flask over it) even if you won't do anything with them. That's why I didn't use Docker client here but called Dockerhub endpoint directly. With current solution even after `docker image prune` command running this script takes several seconds, leaves no traces in Docker on my machine and just says that all images were already built.

When this code is used inside Docker builder it's OK as it Docker would pull 1 image itself (it is caused by `FROM` statement in model image `Dockerfile`).

Reply:
my thought that side-effect of this function was "we have base server image" locally, either by building or pulling it.